src/plugins/profiling/server/routes/functions.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { schema, TypeOf } from '@kbn/config-schema';
+import type { ElasticsearchClient, IRouter, Logger } from 'kibana/server';
+import type { DataRequestHandlerContext } from '../../../data/server';
+import { getRoutePaths } from '../../common';
+import { createTopNFunctions } from '../../common/functions';
+import { logExecutionLatency } from './logger';
+import { newProjectTimeQuery, ProjectTimeQuery } from './query';
+import { downsampleEventsRandomly, findDownsampledIndex } from './downsampling';
+import {
+  mgetExecutables,
+  mgetStackFrames,
+  mgetStackTraces,
+  searchEventsGroupByStackTrace,
+  searchStackTraces,
+} from './stacktrace';
+import { getClient } from './compat';
+
+async function queryTopNFunctions(
+  logger: Logger,
+  client: ElasticsearchClient,
```

Comment:
Using the same names as at other places, like `esClient` and `targetSampleSize` makes it easier to follow, IMO.

Reply:
As I was working on other features, I also noticed naming inconsistencies, opportunities to reduce duplication, linting warnings, and unidiomatic TypeScript. I've been cleaning this up incrementally as I go along, but making the work explicit by creating a cleanup issue makes sense as we get closer to MVP and prepare to merge into the main Kibana repo.